docker-compose.yml

```diff
@@ -2,61 +2,63 @@ version: "3"
 
 services:
   frontend:
-    build: ./frontend
+    image: node:12.18
     container_name: 'ecamp3-frontend'
     ports:
       - '3000:3000'
+    stdin_open: true
+    tty: true
     volumes:
       - ./frontend:/app:delegated
-      - node_modules:/app/node_modules
-      - ./frontend/node_modules:/node_modules-copy-to-host
+      - npm-cache:/root/.npm
+      - cypress-cache:/root/.cache/Cypress
+      - node-modules-cache:/app/node_modules/.cache
+    working_dir: /app
+    command: ./docker-setup.sh
```

Comment:
Ich hätte hier weiterhin `entrypoint` statt `command` verwendet, aber kann man wohl diskutieren was korrekt ist.
Falls docker-setup.sh in Zukunft mal Argumente akzeptiert, ist es einfacher mit entrypoint.
Falls der Frontend-Container in Zukunft mal mit einem anderen Befehl laufen können soll, ist es einfacher mit command.

Ich denke ich würde es auf jeden Fall bei Backend, Composer und Frontend gleich machen, ausser es gibt gute Gründe dagegen.

Reply:
Bin nicht ganz sicher of ich die Unterschiede zwischen `command` und `entrypoint` genügend verstehe. Der `command` war für mich einfacher zum Überschreiben via `docker-compose run`, wie es z.B. im Makefile aufgerufen wird aber auch zum Starten von anderen scripts wie z.B. 'npm run test:unit'